File: src/main/java/com/github/liebharc/queryenrichment/FilterUtils.java
package com.github.liebharc.queryenrichment;

import java.util.List;

public class FilterUtils {

    public static String getSqlCriteria(List<QueryFilter> criteria) {
        final StringBuilder result = new StringBuilder();
        boolean isFirst = true;
        for (QueryFilter criterion : criteria) {

            if (!isFirst) {
                result.append(" and ");
            }

            result.append(criterion.toPlaceHolderString());
            isFirst = false;
        }

        return result.toString();
    }

    private FilterUtils() {

    }
}

File: src/main/java/com/github/liebharc/queryenrichment/FilterUtils.kt
package com.github.liebharc.queryenrichment

object FilterUtils {

    fun getSqlCriteria(criteria: List<QueryFilter>): String {
        val result = StringBuilder()
        var isFirst = true
        for (criterion in criteria) {

            if (!isFirst) {
                result.append(" and ")
            }

            result.append(criterion.toPlaceHolderString())
            isFirst = false
        }

        return result.toString()
    }
}
